modules/openxoutstreamBidAdapter.js

```diff
@@ -0,0 +1,223 @@
+  return {
+    method: 'GET',
+    url: url,
+    data: queryParams,
+    payload: {'bids': bids}
+  };
+}
+
+function handleVastResponse(response, serverResponse) {
+  const body = response.body
+  let bidResponses = [];
+  if (response !== undefined && body.vastUrl !== '' && body.pub_rev && body.pub_rev > 0) {
+    let vastQueryParams = parse(body.vastUrl).search || {};
+    const openHtmltag = '<html><head></head><body>';
+    const closeHtmlTag = '</body></html>';
+    const sdkScript = createSdkScript().outerHTML;
+    const placementDiv = createPlacementDiv();
+    placementDiv.dataset.pID = PUBLISHER_ID;
+    const placementDivString = placementDiv.outerHTML;
+    const adResponse = getTemplateAdResponse(body.vastUrl, PLACEMENT_ID);
+    const adResponseString = JSON.stringify(adResponse);
+    const ymAdsScript = '<script type="text/javascript"> window.__ymAds =' + adResponseString + '</script>';
+
+    let bidResponse = {};
+    bidResponse.requestId = serverResponse.bids[0].bidId;
+    bidResponse.bidderCode = BIDDER_CODE;
+    bidResponse.netRevenue = NET_REVENUE;
+    bidResponse.currency = CURRENCY;
+    bidResponse.cpm = Number(body.pub_rev) / 1000;
+    bidResponse.creativeId = body.adid;
+    bidResponse.height = body.height;
+    bidResponse.width = body.width;
+    bidResponse.vastUrl = body.vastUrl;
+    bidResponse.ttl = TIME_TO_LIVE;
+    bidResponse.mediaType = BANNER;
+    bidResponse.ad = openHtmltag + placementDivString + ymAdsScript + sdkScript + closeHtmlTag;
+
+    response.ph = vastQueryParams.ph;
+    response.colo = vastQueryParams.colo;
+    response.ts = vastQueryParams.ts;
+    bidResponses.push(bidResponse);
+  }
+  return bidResponses;
+}
+registerBidder(spec);
+
+// HELPER FUNCTIONS
+function createSdkScript() {
+  const script = document.createElement('script');
+  script.innerHTML = YM_SCRIPT;
+  return script;
+}
+function createPlacementDiv() {
+  const div = document.createElement('div');
+  div.id = `ym_${PLACEMENT_ID}`;
+  div.classList.add('ym');
+  div.dataset.lfid = `${LICENSED_FORMAT_ID}`;
+  return div
+}
+
+/**
+ * Create a nativeplay template with the placement id and vastURL.
+ * @param vastUrl
+ * @param placementId
+ */
+const getTemplateAdResponse = (vastUrl, placementId) => {
+  return {
+    availability_zone: 'us-east-1a',
+    data: [
+      {
+        ads: [
+          {
+            actions: {},
+            ad_id: AD_ID,
+            adv_id: `${AD_ID}`,
+            configurables: {
+              cta_button_copy: 'Learn More',
+              vast_click_tracking: 'true',
+              vast_url: vastUrl,
+            },
+            cr_id: CR_ID,
+            rti: '1',
+          }
+        ],
+        column_count: 1,
+        configs: {
+          allowable_height: '248',
+          header_copy: 'You May Like',
+          ping: 'true',
+        },
+        creative_format_id: 40,
+        css: '',
+        placement_id: placementId,
```

Comment:
NIT: If we're using constants for a few variables, might as well use constants for all the variables. No point passing in `placementId` as a param as it is always hard coded. Use the constant for `placementId` here.

Reply:
ya that makes sense. updated.